### ISSUE TITLE
Uncaught Error: socket hang up (MainRenderer:182)

### ISSUE THREAD

**Issue by dvdkhlng** [ubports/ubports-installer#196]
This happens when running the ubports-installer under Devuan Ascii (which is a derivative of Debian stable/stretch).

If I continue clicking on "try again", I get back to the UBports Installer window, but it does not seem to fully work:  clicking on "Select device manually" does not show any action and there is no other button to proceed.

Automatically generated error report 
Version: 0.1.12-beta 
Host OS: linux 
Host Dist: Debian 2.0
Host Arch: x64 
Node: v8.2.1 

Error log: http://paste.ubuntu.com/p/c5kmp45TQ3/

**Comment by gsilvapt** [comment]
Thanks for reporting. This is a new one!

Can you provide more details to your environment? Namely what device you're using and what `fastboot devices` and `adb devices` return after you plug your device? No need to give the full output, I just want to confirm the device is recognized and is not failing due to the unstable nature of these USB cables.

**Comment by dvdkhlng** [comment]
Hi Gustavo,

I suspected my installation was too old and upgraded to Devuan Ascii
(which is derived from Debian Stretch (stable)).

But same error still happening.  Is there any implicit dependency on
Systemd in ubports-installer?  There is no systemd in Devuan.

Adb reports:

$ adb devices -l
List of devices attached
* daemon not running. starting it now on port 5037 *
* daemon started successfully *
ML000362               device usb:1-4.4 product:Aquaris_M10_FHD model:Aquaris_M10_FHD_Ubuntu_Edition device:frieza

'fastboot devices' does not return anything.

My tablet is currently running the Ubports stable channel (15.04)
(installed via ubports-installer running on a Ubuntu 17.10 computer)

I just ran:

  strace -f -o /tmp/strace.log /opt/ubports-installer/ubports-installer

But trace does not show any obvious problems.  Anything special to look
out for?

This does more look like a problem with the GUI/run-time than with the
actual low-level interfacing functionality.

cheers,

David
